### ISSUE TITLE
unrecognized argument for puts: -1 [command-args]

### ISSUE THREAD

**Issue by vries** [nmoroze/tclint#108]
```
$ cat test.tcl
#!/usr/bin/tclsh
puts -1
$ ./test.tcl
-1
$ tclint test.tcl
test.tcl:3:1: unrecognized argument for puts: -1 [command-args]
$
```

The test-case is valid TCL.

There's a comment in check_arg_spec that explains why this is reported:
```
       # To facilitate better error messages, we expect that switches are always
       # specified as BareWords that start with "-" or ">". This lets us throw an
       # error when a switch-like thing doesn't match any supported arguments,
       # rather than counting it towards the positional arguments (which usually
       # ends up in a vague "too many arguments" error). To make tclint interpret a
       # switch-like word as a positional argument, users should wrap it in "", and
       # any switches should be BareWords.

```

So, according to the comment, tclint users are expected to write:
`puts "-1"`

Beside the better error message argument,  I suppose this is a trade-off to be able to detect say:
```
$ cat test.tcl
#!/usr/bin/tclsh
puts -nonewlin
$ tclint test.tcl
test.tcl:3:1: unrecognized argument for puts: -nonewlin [command-args]

```

This choice is not documented in violations.md at command-args, so users:
-  have no way of knowing how to fix this other than trial and error, or reading the tclint source code, and
-  may [conclude that tclint has a bug](https://inbox.sourceware.org/gdb-patches/87h5x7oh55.fsf@tromey.com/).

Also the error message could be improved,  it could say unrecognized switch argument, that could give a hint.

We could make this behavior configurable, to make tclint accept this valid TCL instead.

**Comment by tromey** [comment]
It seems to me that for builtins like `append`, the linter could know that there aren't any switch-style arguments.

**Comment by nmoroze** [comment]
Thanks for the report, this and #106 are related bugs that are consequences of an oversight I made during some recent(ish) refactoring. 

I think the proposed patch is a good solution for now. If you submit this in a PR I'm happy to merge it!
